endpointComponent/endpoint.c

```diff
@@ -0,0 +1,96 @@
+#include "interfaces.h"
+#include "legato.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <termios.h>
+#include "connectivity/conn_http.h"
+#include "utils/crypto_utils.h"
+#include "utils/serializer.h"
+#include "utils/tryte_byte_conv.h"
+#include "utils/uart_utils.h"
+#include "utils/utils.h"
+
+#define MSG "%s:%s"
+
+COMPONENT_INIT {
+  uint8_t ciphertext[1024] = {0}, iv[16] = {0}, raw_msg[1000] = {0};
+  uint32_t raw_msg_len = 1 + ADDR_LEN + 20, ciphertext_len = 0, msg_len;
+  char tryte_msg[1024] = {0}, msg[1024] = {0}, addr[ADDR_LEN + 1] = ADDRESS, next_addr[ADDR_LEN + 1] = {0};
+  srand(time(NULL));
+
+  // init uart for modem
+  int fd = uart_init();
+  if (fd < 0) {
+    LE_ERROR("Error in initializing UART\n");
+  }
+
+  // create log file
+  if (log_address(next_addr) != 0) {
+    LE_ERROR("log address failed\n");
+  }
+
+  char *response = NULL;
+  time_t timer;
+  char time_str[26];
+  struct tm *tm_info;
+
+  fd_set rset;
+  struct timeval tv;
+  tv.tv_sec = 0;
+  tv.tv_usec = 500;
+  // TODO:change to none blocking I/O descriptor
+  while (true) {
+    FD_ZERO(&rset);
+    FD_SET(fd, &rset);
+    select(fd + 1, &rset, NULL, NULL, &tv);
+
+    if (FD_ISSET(fd, &rset)) {
+      time(&timer);
+      tm_info = localtime(&timer);
+      strftime(time_str, 26, "%Y-%m-%d %H:%M:%S", tm_info);
+      printf("%s\n", time_str);
+      gen__rand_trytes(ADDR_LEN, next_addr);
+
+      response = uart_read(fd);
+      // real transmitted data
+      snprintf((char *)raw_msg, raw_msg_len, MSG, next_addr, response);
+      LE_DEBUG("Raw_msg:%s\n", raw_msg);
+      uint8_t private_key[AES_KEY_SIZE] = {0};
+      uint8_t id[IMSI_LEN + 1] = {0};
+
+      if (get_aes_key(private_key) != 0) {
+        LE_ERROR("%s\n", "get aes key error");
```

Comment:
Are we able to use a macro to wrapper `LOG_ERROR` to wrap `LE_ERROR`, since we may need to run ta-endpoint on other platform (i.e. testing on linux on laptop).
Maybe we could open another issue for this task

Reply:
Create a new issue for these question?

Reply:
> Are we able to use a macro to wrapper `LOG_ERROR` to wrap `LE_ERROR`, since we may need to run ta-endpoint on other platform (i.e. testing on linux on laptop).
> Maybe we could open another issue for this task

issue at #28